### ISSUE TITLE
Resolved data type is wrong

### ISSUE THREAD

**Issue by peter-gribanov** [kalessil/phpinspectionsea#1522]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.3 |
| Language level | PHP 7.3                        |

May be related #1510

Test code

```php
function test(string $root, string $path): string
{
    $path = parse_url($path, PHP_URL_PATH) ?: $path;

    return $root.$path;
}
```

Current behaviour
--

Wrong type resolved. There the `$path` variable will never be a boolean.

![image](https://user-images.githubusercontent.com/1954436/75462567-808dd780-5995-11ea-9f51-fa83d53a473f.png)

Expected behaviour
--

no error

Environment details
---

```
PhpStorm 2019.3.2
Build #PS-193.6015.48, built on January 22, 2020
```

**Comment by kalessil** [comment]
Fixed!

**Comment by peter-gribanov** [comment]
@kalessil this problem is still relevant.

![image](https://user-images.githubusercontent.com/1954436/82339226-79a3bd80-99f6-11ea-8c1a-12194729a3af.png)

PhpStorm 2020.1.1
Build #PS-201.6668.153, built on April 13, 2020

Php Inspections ​(EA Extended)
4.0.4.1

**Comment by peter-gribanov** [comment]
Do i understand correctly that a solution can be expected in release 4.0.5?

**Comment by ea-inspections-team** [comment]
@peter-gribanov: yes, see the "milestone" details on the right side-pannel of this issue.
